src/app/icd/ICDEventManager.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/icd/ICDEventManager.h>
+
+using namespace chip::DeviceLayer;
+
+namespace chip {
+namespace app {
+
+/**
```

Comment:
please move doc-comments into headers as that is what people usually open for looking at the API. If duplicated, remove the cpp copy.

also nit: make these actually useful and remove things that are not needed (e.g. "IDCManager *" is obviously a pointer to an ICDManager and `remote CHIP_ERROR` is obvious in the function signature and does not need commenting if nothing useful can be said about it.

Reply:
done